Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,700 @@
+    [ContinuousConstSMul R β] : MulActionWithZero R C_c(α, β) :=
+  Function.Injective.mulActionWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance instModule [AddCommMonoid β] [ContinuousAdd β] {R : Type*} [Semiring R] [Module R β]
+    [ContinuousConstSMul R β] : Module R C_c(α, β) :=
+  Function.Injective.module R ⟨⟨_, coe_zero⟩, coe_add⟩ DFunLike.coe_injective coe_smul
+
+-- TODO: solve type class instance problem
+-- related with the assumption `[AddCommMonoid β] [ContinuousAdd β]` in `Add`?
+-- instance instNonUnitalNonAssocSemiring [NonUnitalNonAssocSemiring β] [TopologicalSemiring β] :
+--     NonUnitalNonAssocSemiring C_c(α, β) :=
+--   DFunLike.coe_injective.nonUnitalNonAssocSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+-- instance instNonUnitalSemiring [NonUnitalSemiring β] [TopologicalSemiring β] :
+--     NonUnitalSemiring C_c(α, β) :=
+--   DFunLike.coe_injective.nonUnitalSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+-- instance instNonUnitalCommSemiring [NonUnitalCommSemiring β] [TopologicalSemiring β] :
+--     NonUnitalCommSemiring C_c(α, β) :=
+--   DFunLike.coe_injective.nonUnitalCommSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalNonAssocRing [NonUnitalNonAssocRing β] [TopologicalRing β] :
+    NonUnitalNonAssocRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instNonUnitalRing [NonUnitalRing β] [TopologicalRing β] : NonUnitalRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub (fun _ _ => rfl)
+    fun _ _ => rfl
+
+instance instNonUnitalCommRing [NonUnitalCommRing β] [TopologicalRing β] :
+    NonUnitalCommRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+-- instance instIsScalarTower {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+--     [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [IsScalarTower R β β] :
+--     IsScalarTower R C_c(α, β) C_c(α, β) where
+--   smul_assoc r f g := by
+--     ext
+--     simp only [smul_eq_mul, coe_mul, coe_smul, Pi.mul_apply, Pi.smul_apply]
+--     rw [← smul_eq_mul, ← smul_eq_mul, smul_assoc]
+
+-- instance instSMulCommClass {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+--     [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [SMulCommClass R β β] :
+--     SMulCommClass R C_c(α, β) C_c(α, β) where
+--   smul_comm r f g := by
+--     ext
+--     simp only [smul_eq_mul, coe_smul, coe_mul, Pi.smul_apply, Pi.mul_apply]
+--     rw [← smul_eq_mul, ← smul_eq_mul, smul_comm]
+
+end AlgebraicStructure
+
+section Uniform
+
+variable [UniformSpace β] [UniformSpace γ] [Zero γ]
+variable [FunLike F β γ] [CompactlySupportedContinuousMapClass F β γ]
+
+lemma zero_at_infty_of_hasCompactSupport [TopologicalSpace β] [Zero β]
```

Comment:
This lemma only needs topological spaces, not uniform spaces, so it's a little bit strange to have it in the section `Uniform`. Also, this lemma means that you can (should?) instantiate a `ZeroAtInftyContinuousMapClass` instance here.

Reply:
OK, I will register an instance.